port of image_projection to ros-jazzy

FrederikBark opened the pull request:
Current state of the jazzy port. We are currently preparing test on our new robot.

FrederikBark committed:
port of image_projection to ros-jazzy

FrederikBark committed:
removed unused cfg folder

FrederikBark committed:
changed runtime dir to lib

FrederikBark committed:
change runtime location to lib/${PROJECT_NAME}

Joschi3 committed:
CMakeList modification

Excelsius314 committed:
Fix img stamps

StefanFabian committed:
Formatting. Added check to not publish a new image if none of the source images have changed.
Probably improved performance of mapping computation by quite a bit.

StefanFabian committed:
Added logging and fixed always_recompute_mapping_ only recomputing always when it is false.

StefanFabian committed:
Formatting and debug log.

StefanFabian committed:
Added more logging.

StefanFabian committed:
More debug logging.

StefanFabian committed:
More debug logging.

StefanFabian committed:
More debug logging.

StefanFabian committed:
Check if camera info is received before using a camera.

StefanFabian committed:
Removed debug logging.

StefanFabian committed:
Added option to set log level when launching.

FrederikBark committed:
added missing build_type export

lreiher commented:
@FrederikBark Hi! What's the status of the port to ROS 2 Jazzy? We are looking forward to using it on our vehicles.

I can build the package on this branch along with [hector_ros2_utils](https://github.com/tu-darmstadt-ros-pkg/hector_ros2_utils) and [kalibr_image_geometry](https://github.com/tu-darmstadt-ros-pkg/kalibr_image_geometry/tree/feature/jazzy-port), but the node crashes immediately at runtime.

```bash
dockeruser@7ea1538630e2:~/ws$ ros2 run image_projection periodic_image_projection_standalone_node --ros-args -p projection_type:=image_projection_plugins::EquirectangularProjection -p base_frame:=base_link
[INFO] [1758224941.590803106] [periodic_image_projection_node]: Save folder:
[ros2run]: Segmentation fault
```

```bash
dockeruser@7ea1538630e2:~/ws$ ros2 run image_projection image_projection_server_standalone_node
# this doesn't crash and there is a living node, but it doesn't have any logs or subscriptions/publishers
```

FrederikBark committed:
updated dynamic parameter configuration so that image resolutions can be
adapted on the fly
fixed missing cmaera infos due to parameter namespace conflict

FrederikBark committed:
added set transform which allows to use base pose and transform for
virtual camera

FrederikBark committed:
add init for virtual transform

FrederikBark committed:
changed no subscriber info to debug instead of info log level

FrederikBark committed:
bug fixes and linting

FrederikBark committed:
set info message "No new images received" to debug instead of info

FrederikBark committed:
added std move and unique ptr for publishers

FrederikBark committed:
updated image projection and fixed invalid unique ptr

FrederikBark committed:
added default enable_pub_plugins value to avoid unnecessary topics
